### ISSUE TITLE
Use WorkManager in telemetry library

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#160]
JobScheduler is broken in certain API levels. WorkManager fixes all that and makes the library backwards compatible.

**Comment by pocmo** [comment]
At the moment 1.0.0-**alpha05** seems to be the latest version.

**Comment by pocmo** [comment]
WONTFIX: We are working actively on the new library "glean" and with that we won't need to make this update in the old telemetry library. We can't merge this anytime soon until *all* apps using telemetry have migrated to AndroidX and this is very unlikely to happen any time soon.
